library/common/network/proxy_settings.h

```diff
@@ -62,6 +70,9 @@ struct ProxySettings {
     if (address_ != nullptr) {
       return address_->asString();
     }
+    if (!hostname_.empty()) {
```

Comment:
we need to update the implementation of `==` operator in this file too. Some logic in connectivity_manager does depend on it being accurate.